truss-chains/truss_chains/deployment/code_gen.py

```diff
@@ -458,13 +467,15 @@ def _gen_predict_src(chainlet_descriptor: definitions.ChainletAPIDescriptor) ->
         maybe_await = ""
     run_remote = chainlet_descriptor.endpoint.name
     # See docs of `pydantic_set_field_dict` for why this is needed.
-    args = "**stub.pydantic_set_field_dict(inputs)"
+    args = "**utils.pydantic_set_field_dict(inputs)"
     parts.append(
         _indent(f"result = {maybe_await}self._chainlet.{run_remote}({args})", 2)
     )
     if chainlet_descriptor.endpoint.is_streaming:
         # Streaming returns raw iterator, no pydantic model.
-        parts.append(_indent("return result"))
+        # This needs to be nested inside the `trace_parent` context!
+        parts.append(_indent("async for chunk in result:", 2))
```

Comment:
just to be clear, the problem with just returning the result is that the actual generator executes outside of the context of the trace parent right?